### PR TITLE
Add basic handheld GPS

### PR DESCRIPTION
**Pull request by Fishfish458**
## About the PR 
This is an attempt at the most absolute basic form of GPS 

**Screenshots**

h d

**Changelog**

:cl:
- add: Added handheld GPS devices which can be found in the autolathe

### CODE REVIEW

**Comment by Elijahrane** [comment]
Any possibility this could be built into the handheld crew monitor UI?

**Comment by mirrorcult** [comment]
> Any possibility this could be built into the handheld crew monitor UI?

spacemen frothing at the mouth at the thought of needing two devices to do something

**Comment by moonheart08** [comment]
Don't see a reason to not just use two devices.

**Comment by metalgearsloth** [comment]
BoundUI and networking seems like overkill when you could just use IItemStatus considering the client already knows its position...

**Comment by Fishfish458** [comment]
> BoundUI and networking seems like overkill when you could just use IItemStatus considering the client already knows its position...

I will look into that right now

**Comment by moonheart08** [comment]
Personally I think the networking is fine, especially if GPS's transmitters are made more tangible.

**Comment by metalgearsloth** [comment]
I thought we already had bandwidth issues...

**Comment by moonheart08** [comment]
> I thought we already had bandwidth issues...

would sending a few numbers for state every second or two for a select set of users cause bandwidth problems?

**Comment by Fishfish458** [comment]
OK, made some changes so that this now uses an iitemstatus for basically everything.

changed

File: Content.Client/GPS/HandheldGPSComponent.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+            protected override void FrameUpdate(FrameEventArgs args)
+            {
+                base.FrameUpdate(args);
+
+                UpdateDif += args.DeltaSeconds;
+                if (UpdateDif < _parent.UpdateRate)
+                    return;
+
+                UpdateDif = 0f;
```
Comment (metalgearsloth) [inline Content.Client/GPS/HandheldGPSComponent.cs:43]:
```suggestion
                UpdateDif -= _parent.UpdateRate;
```

File: Content.Client/GPS/HandheldGPSComponent.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+                UpdateGPSDetails();
+            }
+
+            public void UpdateGPSDetails()
+            {
+                _entMan.TryGetComponent<TransformComponent>(_parent.Owner, out TransformComponent? transComp);
+
+                string posText = "Error";
+                if (transComp != null)
```
Comment (metalgearsloth) [inline Content.Client/GPS/HandheldGPSComponent.cs:53]: Tryget returns a bool so you don't need to check for null.

File: Resources/Prototypes/Recipes/Lathes/tools.yml [hunk 3]
```diff
@@ -134,3 +134,13 @@
...
     Plastic: 250
+
+- type: latheRecipe
+  id: HandheldGPSBasic
+  icon: Objects/Devices/gps.rsi/icon.png
+  result: HandheldGPSBasic
+  completetime: 2000
+  materials:
+    Steel: 800
+    Plastic: 400
```
Comment (metalgearsloth) [inline Resources/Prototypes/Recipes/Lathes/tools.yml:145]: Isn't plastic very rare? Not sure how valuable the gps is compared to what it requires to get but I guess we'll see.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Delete empty server system

File: Resources/Prototypes/Recipes/Lathes/tools.yml (hunk 3 above)
Comment (Fishfish458) [reply Resources/Prototypes/Recipes/Lathes/tools.yml:145]: I'll just remove the plastic since the steel and glass input enough is a decent bit.

Comment (Fishfish458) [reply Resources/Prototypes/Recipes/Lathes/tools.yml]: > Delete empty server system

Yeah, I feel real dumb for that one. overlooked it.
Made the changes and everything is still working, thanks.

**Pull request merged**